src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+price is stored in listing public data in an object with two keys:
+`amount` and `currency`. The `amount` attribute holds the price in sub
+units whereas `currency` holds the currency code. Like this for a
+cleaning fee of 20€:
+
+```js
+publicData: {
+  cleaningFee: { amount: 2000, currency: 'EUR' }
+}
+```
+
+Begin with making a few changes to the `util/types.js` and `en.json`
+files (or some other transaltion file in case English is not the
+language of your marketplace). In `util/types.js` add a new line item
+code and add it to the array of line items:
+
+```js
+export const LINE_ITEM_CLEANING_FEE = 'line-item/cleaning-fee';
+
+const LINE_ITEMS = [
+  LINE_ITEM_NIGHT,
+  LINE_ITEM_DAY,
+  LINE_ITEM_UNITS,
+  LINE_ITEM_CUSTOMER_COMMISSION,
+  LINE_ITEM_PROVIDER_COMMISSION,
+  LINE_ITEM_CLEANING_FEE,
+];
+```
+
+To the `en.json` translation file, add the following new translation
+keys:
+
+```json
+"BookingDatesForm.cleaningFee": "Add cleaning",
+"BookingBreakdown.cleaningFee": "Cleaning fee",
+```
+
+In order to add the cleaning fee option to the booking form on listing
+page, first add the following lines to the bottom of the imports in the
+`BookingPanel` component:
+
+```js
+import { types as sdkTypes } from '../../util/sdkLoader';
+const { Money } = sdkTypes;
+```
+
+In the same `BookingPanel` component add the following code to the top
+of the component, right under where the props object is destructured. It
+will resolve the the price of the cleaning fee if one is defined for the
+current listing:
+
+```js
+const cleaningFeeData = listing.attributes.publicData.cleaningFee;
+const { amount: cleaningAmount, currency: cleaningCurrency } =
+  cleaningFeeData || {};
+const cleaningFee =
+  cleaningAmount && cleaningCurrency
+    ? new Money(cleaningAmount, cleaningCurrency)
+    : null;
+```
+
+Also in `BookingPanel` add a submit handler function that converts a
+checkbox form value into the actual cleaning fee of this listing:
+
+```js
+const handleSubmit = values => {
+  const selectedCleaningFee =
+    values &&
+    values.additionalItems &&
+    values.additionalItems[0] === 'cleaningFee'
+      ? cleaningFee
+      : null;
+  onSubmit({
+    ...values,
+    cleaningFee: selectedCleaningFee,
+  });
+};
+```
+
+Pass `cleaningFee` and `handleSubmit` to the `BookingDatesForm`
+component as follows:
+
+```js
+<BookingDatesForm
+  className={css.bookingForm}
+  submitButtonWrapperClassName={css.bookingDatesSubmitButtonWrapper}
+  unitType={unitType}
+  onSubmit={handleSubmit}
+  price={price}
+  isOwnListing={isOwnListing}
+  timeSlots={timeSlots}
+  fetchTimeSlotsError={fetchTimeSlotsError}
+  cleaningFee={cleaningFee}
+/>
+```
+
+In the `BookingDatesForm` component file import the `FieldCheckbox`
+component and the `formatMoney` utility function:
+
+```js
+import {
+  Form,
+  PrimaryButton,
+  FieldDateRangeInput,
+  FieldCheckbox,
+} from '../../components';
+import { formatMoney } from '../../util/currency';
```

Comment:
Swap the order of these two imports.

Reply:
How come?

Reply:
IRL: import util functions before internal components to prevent circular dependencies.